multi-layer.md

```diff
@@ -0,0 +1,130 @@
+* Only `XRWebGLLayer` can be set to the `baseLayer`; only new layer types (`XRLayerProjection`, `XRLayerQuad`, etc) can be set as elements of `layers`.
+* The types of supported layers could be asked from [XRSession](https://immersive-web.github.io/webxr/#xrsession) by using `isLayerSupported(layer_class_name)` method;
+* Layers are drawn in the same order as they are specified in via `XRSession/updateRenderState`, with the 0th layer drawn first. Layers are drawn with a "painter’s algorithm," with each successive layer potentially overwriting the destination layers whether or not the new layers are virtually closer to the viewer. There is no depth testing between the layers. Layer's visibility is controlled by presence (or absence) of the layer in the sequence. To hide or show a layer - call [updateRenderState](https://immersive-web.github.io/webxr/#dom-xrsession-updaterenderstate) and provide a new sequence of layers.
+
+* Add common properties to [XRLayer](https://immersive-web.github.io/webxr/#xrlayer-interface) (such as chromatic aberration or alpha blending flags):
+* * `chromaticAberrationCorrection` - boolean, enables optical chromatic aberration correction for the layer when not done by default; chromatic aberration correction improves visual fidelity of the layer's rendering, but it might be computationally expensive; this flag may be ignored by the browser, if this feature is not supported.
+* * `blendTextureSourceAlpha` - boolean, enables the layer’s texture alpha channel. If enabled, then the layer composition uses the alpha channel to modulate the blending of the layer's texture against the destination.  The texture color channels must be encoded with premultiplied alpha. If the bit is not enabled, then each texel is treated as if alpha channel has value 1.0.
+This feature must be supported if multiple layers are supported; otherwise, it is ignored.
+* * `void requestUpdate()` - the method that should be called to indicate the layer's content changed; by default, browser should assume that the content of the layers (image source) is not updated.
+
+> **TODO** Verify with WebGL folks the alpha-premultiply stuff.
+
+See the Appendix for the proposed IDL.
+
+
+## Code examples
+
+### Checking layers' types and maximum amount of layers supported and creating layers:
+```javascript
+// session is already created
+let maxLayerCount = session.getMaxLayerCount();
+let newRenderState = new Object;
+if (maxLayerCount == 0) {
+  // we use baseLayer
+  newRenderState.baseLayer = new XRWebGLLayer(session, gl);
+} else {
+  newRenderState.layers = new Array();
+  newRenderState.layers[0] = new XRLayerProjection(session, gl);
+  if (maxLayerCount > 1 && session.isLayerSupported("XRLayerQuad"))
+    newRenderState.layers[1] = new XRLayerQuad(session, gl);
+  }
+}
+session.updateRenderState(newRenderState);
+
+```
+
+### Setting common layers properties
+```javascript
+  if (maxLayerCount > 1 && session.isLayerSupported("XRLayerQuad"))
```

Comment:
I'm not sure about the boolean "isLayerSupported" - that's a hard question for the implementation to answer if it doesn't yet know what flags will be set on the layer and what size they'll be.

Alternatively, would it make sense to have a layer configuration attribute set that's provided to both the query function and the layer initializer? That way, there's less risk of inconsistency, and the implementation could potentially respond with "I could do this, but it's unlikely to hit target framerates". That could either be on a layer-by-layer basis, or potentially as a "here's my proposed overall layer configuration" all-at-once call.

Reply:
Sounds good, will try to make the change